part-spec/Registers.json

```diff
@@ -0,0 +1,88 @@
+                "type": "string"
+            },
+            "registerLongName": {
+                "description": " Full Name of a register",
+                "type": "string"
+            },
+            "registerAddressOffset": {
+                "description": "Address of a register",
+                "type": "number",
+                "comment": "Hex numeral system"
+            },
+            "registerSize": {
+                "description": "register Size",
+                "type": "number",
+                "comment": "units of bytes"
+            },
+            "registerType": {
+                "description": "Type of a register",
+                "type": "string",
+                "comment": "examples include but are not limited to data registers, address registers"
+            },
+            "registerResetValue": {
+                "description": "Reset value of a register",
+                "type": "number",
+                "comment": "Hex numeral system"
+            },
+            "registerValue": {
+                "description": "Value of a register",
+                "type": "number",
+                "comment": "Hex numeral system"
+            },
+            "registerAccessType": {
+                "description": " Access type ( Read, Write) of a Register",
+                "type": "string",
+                "comment": " examples include but are not limited to Read Only, WRite, Read Write"
+            },
+            "registerBitField": {
+                "description": " Describes the bit fileds in the register",
+                "$ref": "#/defs/registerBitField"
+            }
+        }
+    },
+    "$defs": {
+        "registerBitField": {
+            "type": "object",
+            "properties": {
+                "bitFieldName": {
+                    "description": "Name of the bit field",
+                    "type": "string"
+                },
+                "bitFieldLongName": {
+                    "description": "Long Name of the bit field",
+                    "type": "string"
+                },
+                "bitFieldDescription": {
+                    "description": "Describes the bit field",
+                    "type": "string"
+                },
+                "bitFieldRange": {
+                    "description": "Range of the bit field",
+                    "type": "string",
+                    "comment": "single bit notation or Highest bit:lowest bit notation"
+                },
+                "bitFieldResetValue": {
+                    "description": "Reset value of a bit field",
+                    "type": "number",
+                    "comment": "Hex numeral system",
+                },
+                "bitFieldAccessType": {
+                    "description": " Access type ( Read, Write) of a bit field",
+                    "type": "string",
+                    "comment": " examples include but are not limited to Read Only, WRite, Read Write"
```

Comment:
whatever we come up with above, we should use here. Also are there write-only registers? 

(silly joke: https://web.archive.org/web/20120316141638/http://www.national.com/rap/files/datasheet.pdf)